pkg/liqo-controller-manager/namespaceOffloading-controller/namespaceoffloading_lifecycle.go

```diff
@@ -47,9 +48,21 @@ func (r *NamespaceOffloadingReconciler) initialConfiguration(ctx context.Context
 	patch := noff.DeepCopy()
 	// 1 - Add NamespaceOffloadingController Finalizer.
 	ctrlutils.AddFinalizer(noff, namespaceOffloadingControllerFinalizer)
-	// 2 - Add empty cluster selector if not specified by the user.
+	// 2 - Add the default ClusterSelector if not specified by the user. The default cluster selector allows creating
+	// remote namespaces on all remote clusters.
+	// The Namespace Offloading resource must always have the ClusterSelector field to correctly enforce
+	// the security policies in the liqo webhook.
 	if noff.Spec.ClusterSelector.Size() == 0 {
-		noff.Spec.ClusterSelector = corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{}}
+		noff.Spec.ClusterSelector = corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{
+			{
+				MatchExpressions: []corev1.NodeSelectorRequirement{{
+					Key:      liqoconst.TypeLabel,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{liqoconst.TypeNode},
+				},
+				},
```

Comment:
I'm not sure about it, it will allow both local and remote clusters or only the remote ones?

And second, the indentation seems strange to me

Reply:
here this MatchExpression says to create remote namespaces on all remote clusters. If pods are scheduled only remotely or even locally, it is decided by the podOffloadingStrategy in the webhook not here. I'm not sure I understand your question correctly. For the second point I will fix it.